### PR TITLE
Import proguard rules from proguard-android.txt

### PR DESCRIPTION
**Pull request by dellis1972**
Context: https://github.com/xamarin/xamarin-android/issues/8337
Fixes: https://github.com/xamarin/xamarin-android/issues/8397

The latest Android sdk no longer ships with `proguard` as a result the file in proguard-android.txt [1] is missing. As a result customers are reporting errors such as

```
Unhandled Exception from source=AndroidEnvironment

Java.Lang.Exception: android.view.View_IOnClickListenerImplementor
    at Java.Interop.JniEnvironment.Types.TryFindClass(String , Boolean )
    at Java.Interop.JniEnvironment.Types.FindClass(String )
    at Java.Interop.JniType..ctor(String )
    at Java.Interop.JniPeerMembers.JniInstanceMethods..ctor(Type )
    at Java.Interop.JniPeerMembers.JniInstanceMethods.GetConstructorsForType(Type )
    at Java.Interop.JniPeerMembers.JniInstanceMethods.StartCreateInstance(String , Type , JniArgumentValue* )
    at Android.Views.View.IOnClickListenerImplementor..ctor()
    at Android.Views.View.__CreateIOnClickListenerImplementor()
    at Java.Interop.EventHelper.AddEventHandler[IOnClickListener,IOnClickListenerImplementor](WeakReference& , Func`1 , Action`1 , Action`1 )
    at Android.Views.View.add_Click(EventHandler )
    at com.glmsoftware.obdnowpros.Fragments.SubscriptionFragment.OnCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
    at AndroidX.Fragment.App.Fragment.n_OnCreateView_Landroid_view_LayoutInflater_Landroid_view_ViewGroup_Landroid_os_Bundle_(IntPtr jnienv, IntPtr native__this, IntPtr native_inflater, IntPtr native_container, IntPtr native_savedInstanceState)
    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPLLL_L(_JniMarshal_PPLLL_L callback, IntPtr jnienv, IntPtr klazz, IntPtr p0, IntPtr p1, IntPtr p2)
    --- End of managed Java.Lang.Exception stack trace ---
```

This is probably because certain code is removed that in the latest version of R8 that used to be left in place.
We should include the contents of this removed file in our own proguard_xamarin.cfg file.

[1] https://android.googlesource.com/platform/sdk/+/refs/heads/main/files/proguard-android.txt

### CODE REVIEW

**Review by jonathanpeppers — COMMENTED** [review]
Should we also remove this line?

https://github.com/xamarin/xamarin-android/blob/19d978bc1b6ee04e6dd4b11ec01d2a9c8f6b3e49/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1943

File: src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets [hunk 1]
```diff
@@ -1940,7 +1940,6 @@ because xbuild doesn't support framework reference assemblies.
     <_ProguardConfiguration Include="$(ProguardConfigFiles)" />
   </ItemGroup>
   <ItemGroup Condition=" '$(ProguardConfigFiles)' == '' ">
-    <_ProguardConfiguration Include="$(_AndroidSdkDirectory)tools\proguard\proguard-android.txt" />
```
Comment (jonathanpeppers) [inline src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets:1943]: Ok, removing this really broke the `LibraryProjectZipWithLint` test. Is there a real issue there?

**Comment by jpobst** [comment]
While I think this is a good change, I'm not convinced it actually fixes the reported issue.

From my testing, this case https://github.com/xamarin/xamarin-android/issues/8337#issuecomment-1753188730 still seems to work on .NET 7 but not .NET 8. I keep investigating but I cannot find any real difference between the 2 that would cause the issue.

**Comment by jonpryor** [comment]
As per discussion on Discord, it looks like https://github.com/xamarin/java.interop/pull/1149 is the actual fix for #8337.

We may still want this change *anyway*, but we should probably re-evaluate this.  Why did Google remove `proguard-android.txt` from the Android SDK?  Did the functionality get moved elsewhere (and we should thus be tracking the new location)?  Is the previous contents "moot" because of other changes in the SDK?  Something else?

I suspect that this will become a .NET 9 task.

**Comment by jpobst** [comment]
> Why did Google remove proguard-android.txt from the Android SDK?

I came across the answer to this earlier in my research:

```
# This file is no longer maintained and is not used by new (2.2+) versions of the
# Android plugin for Gradle. Instead, the Android plugin for Gradle generates the
# default rules at build time and stores them in the build directory.
```

([Source](https://chromium.googlesource.com/android_tools/+/HEAD/sdk/tools/proguard/proguard-android.txt))

Agreed that this is something we will eventually need to further research to ensure we keep up with the Android ecosystem.

**Comment by dellis1972** [comment]
I'm closing this PR as it was not causing the issue.

**Pull request closed without merging**